### PR TITLE
Add support for open id connect token auth

### LINKED ISSUE #35: Add OIDC auth support
Seems this exists in [python-base](https://github.com/kubernetes-client/python-base/blob/master/config/kube_config.py) now, though their version is not entirely functional (story of my life with every open id implementation in existence 😞  )

I am currently working on a pr for this

### PR DESCRIPTION
**Pull request by bpicolo**
Fixes #35.

The one think I wasn't yet able check the actual functionality of is `idp-certificate-authority-data`, but my org will add that sometime soon and then I can give that a shot to see if it works as intended in practice. (It essentially matches what kubectl tries at the moment, though)

Lmk what other tests you'd like to see here. It's based off the kubernetes/python-base version, but I cleaned up a bunch of the code and also fixed a number of bugs. I'm able to successfully auth / run commands against the kubernetes cluster in question with this patch.

Thanks!

### CODE REVIEW

**Comment by bpicolo** [comment]
screen shot 2018-08-01 at 3 29 20 pm

Feels like a moral victory, right? 🙉

File: kubernetes_asyncio/config/kube_config.py [hunk 1]
```diff
@@ -201,6 +212,99 @@ def set_active_context(self, context_name=None):
...
+
+            if response.status != 200:
+                raise ValueError('oidc: failed to query metadata endpoint')
+
+            well_known_data = json.loads(response.data)
+
+            self._perform_oauth_refresh(provider, client, well_known_data)
+
+    def _perform_oauth_refresh(self, provider, client, well_known_data):
+        # It's not optimal that this isn't async, but there's no existing well-supported
```
Comment (tomplus) [inline kubernetes_asyncio/config/kube_config.py]: Yes, and it was a reason why I didn't copy this from the official library. The main goal of this library is to use only asynchronous calls.

**Comment by bpicolo** [comment]
@tomplus There's a functional asyncio for ya ❤️

**Comment by bpicolo** [comment]
@tomplus  this should be ready for re-review when you have a chance

**Comment by tomplus** [comment]
@bpicolo Thanks a lot! I'll check how it works soon...

**Comment by tomplus** [comment]
LGTM

Great job! Thanks for your contribution.

**Pull request merged**